lib/Models/Catalog/Ows/WebMapTileServiceCatalogItem.ts

```diff
@@ -556,6 +537,59 @@ class WebMapTileServiceCatalogItem extends MappableMixin(
     return imageryProvider;
   }
 
+  getTileUrl(
+    layer: WmtsLayer,
+    capabilities: WebMapTileServiceCapabilities,
+    format: string
+  ) {
+    let url: string | undefined = undefined;
+    if (
+      capabilities.OperationsMetadata &&
+      "GetTile" in capabilities.OperationsMetadata
+    ) {
+      const gets = capabilities.OperationsMetadata.GetTile["Get"];
+
+      for (let i = 0; i < gets.length; i++) {
+        let constraints = gets[i].Constraint;
+        if (constraints) {
+          constraints = Array.isArray(constraints)
+            ? constraints
+            : [constraints];
+          const getEncodingConstraint = constraints.find(
+            (element) => element.name === "GetEncoding"
+          );
+
+          const encodings = getEncodingConstraint?.AllowedValues?.Value;
+          if (encodings?.includes("KVP")) {
+            url = gets[i]["xlink:href"];
+          }
+        } else if (gets[i]["xlink:href"]) {
+          url = gets[i]["xlink:href"];
+        }
+      }
+    }
+
+    const resourceUrls: ResourceUrl[] | undefined =
+      !layer.ResourceURL || Array.isArray(layer.ResourceURL)
+        ? layer.ResourceURL
+        : [layer.ResourceURL];
+
+    if (resourceUrls && (this.requestEncoding === "RESTful" || !url)) {
+      for (let i = 0; i < resourceUrls.length; i++) {
```

Comment:
Could `for (const resourceUrl of resourceUrls) {` be used to iterate over the content so the next line can be omitted, or is there some JavaScript-reason to always use index-based iterations in TypeScript?

Same comment for the other for-loops where the index is only used to get the element.